src/content/glossary/index.md

```diff
@@ -873,6 +873,10 @@ Short for "succinct non-interactive argument of knowledge", a SNARK is a type of
   Zero-knowledge rollups
 </DocLink>
 
+### soft fork {#soft-fork}
+
+A temporary divergence in a [blockchain](#blockchain) that occurs when the [consensus rules](#consensus-rules) become stricter. Contrary to a [hard fork](#hard-fork), a soft fork is backwards compatible; upgraded nodes can validate blocks created by nonupgraded nodes as long as they follow the new consensus rules. However, nonupgraded nodes are ultimately encouraged to download the latest version of the blockchain's software to avoid complications related to transactions.
```

Comment:
> However, nonupgraded nodes are ultimately encouraged to download the latest version of the blockchain's software to avoid complications related to transactions.

I'd say this part strays from the job of a glossary (starting to describe the Ethereum specific details instead of simply defining a soft fork at a high level). I think we should delete the last sentence. The rest LGTM 😎👍

Reply:
IMO a soft fork is one that makes non-breaking changes (i.e. nodes that do not implement the fork can still participate in a network of upgraded nodes).

Not keen on "temporary divergence" - temporary suggests it will be reverted eventually, and divergence implies incorrectness or irrelevance (IMO). Also not sure why a soft fork necessarily makes a set of rules "stricter" nor are forks specific to the "consensus rules".

Reply:
@jmcook1186 @minimalsm Sure, I agree with u guys! We sure could make changes to this.

This is a definition that I found online,

A soft fork is a change to the software protocol where only previously valid transaction blocks are made invalid. Because old nodes will recognise the new blocks as valid, a soft fork is backwards-compatible. This kind of fork requires only a majority of the miners upgrading to enforce the new rules, as opposed to a hard fork that requires all nodes to upgrade and agree on the new version.

We can modify and build the definition upon this. I think this covers the concerns raised by @jmcook1186. Let me know ur views as u review this.

Cheers!!

Reply:
Fair enough - I'd equated "consensus rules" with "consensus mechanism" but a soft fork could touch any part of the client software - the specificity seems misleading to me.

Point taken re "stricter", but what about a soft fork that introduces new features? They can be run by clients that update and not by clients that don't without invalidating either - I wouldn't consider any rules to have become stricter in that case, if anything the feature-set, has expanded, updated clients can "do more" and therefore the rule-set has become less strict.